File: game/src/main/java/com/matag/game/turn/action/attach/AttachService.java
package com.matag.game.turn.action.attach;

import com.matag.game.cardinstance.CardInstance;
import com.matag.game.status.GameStatus;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

@Component
public class AttachService {
  private static final Logger LOGGER = LoggerFactory.getLogger(AttachService.class);

  public void attach(GameStatus gameStatus, CardInstance cardInstance, int attachedToId) {
    CardInstance attachTo = gameStatus.findCardByIdFromAnyBattlefield(attachedToId);
    cardInstance.getModifiers().setAttachedToId(attachedToId);
    LOGGER.info("attached {} to {}", cardInstance.getIdAndName(), attachTo.getIdAndName());
  }

  public void detach(CardInstance cardToDestroy, CardInstance attachedCard) {
    attachedCard.getModifiers().unsetAttachedId();
    LOGGER.info("detached {} from {}", attachedCard.getIdAndName(), cardToDestroy.getIdAndName());
  }
}

File: src/main/kotlin/com/matag/game/turn/action/attach/AttachService.kt
package com.matag.game.turn.action.attach

import com.matag.game.cardinstance.CardInstance
import com.matag.game.status.GameStatus
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class AttachService {
    fun attach(gameStatus: GameStatus, cardInstance: CardInstance, attachedToId: Int) {
        val attachTo = gameStatus.findCardByIdFromAnyBattlefield(attachedToId)
        cardInstance.modifiers.attachedToId = attachedToId
        LOGGER.info("attached {} to {}", cardInstance.idAndName, attachTo!!.idAndName)
    }

    fun detach(cardToDestroy: CardInstance, attachedCard: CardInstance) {
        attachedCard.modifiers.unsetAttachedId()
        LOGGER.info("detached {} from {}", attachedCard.idAndName, cardToDestroy.idAndName)
    }

    companion object {
        private val LOGGER: Logger = LoggerFactory.getLogger(AttachService::class.java)
    }
}
